src/actions/auth.js

```diff
@@ -282,7 +287,7 @@ export const createUserProfile = (dispatch, firebase, userData, profile) => {
   if (typeof config.profileFactory === 'function') {
     // catch errors in user provided profileFactory function
     try {
-      profile = config.profileFactory(userData, profile, firebase) // eslint-disable-line no-param-reassign
+      profile = await config.profileFactory(userData, profile, firebase) // eslint-disable-line no-param-reassign
```

Comment:
Is an error thrown if the profileFactory isn't a promise? If so we can add logic to check if the profileFactory is a promise before running await on it

Reply:
From my tests, no errors are thrown if the profileFactory is not a promise. I have tried both examples below (and both work with error or warning):

```
// Async
profileFactory: async (userData, profileData, firebase) => {
  const asyncProfileData = await asyncFunctionToPerformOtherActions(profileData, firebase);
  return {
      ...asyncProfileData
  }
}
  
// Sync
profileFactory: (userData, profileData, firebase) => {
  return {
     ...profileData
  }
}
```

I am happy to add a check anyway if you would like?